### PR TITLE
Travis script that runs engine unit tests and Dart tests of engine APIs

### PR DESCRIPTION
**Pull request by jason-simmons**
The Dart tests were migrated from the flutter/test/engine suite in the
framework repository

### CODE REVIEW

**Comment by jason-simmons** [comment]
@abarth

**Comment by Hixie** [comment]
LGTM

**Comment by eseidelGoogle** [comment]
I don't speak bash well enough to know.  Will test.sh correctly exit on failure when any one of these lines fail?

File: travis/test.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+out/host_debug_unopt/ftl_unittests
+out/host_debug_unopt/synchronization_unittests
+out/host_debug_unopt/wtf_unittests
+
+pushd flutter/testing/dart
+pub get
+popd
+
+for TEST_SCRIPT in flutter/testing/dart/*.dart; do
```
Comment (Hixie) [inline travis/test.sh]: i recommend having a control test that verifies that if one of these fails, it properly gets noticed. See the framework test.sh for (a paranoidly high number of) examples.

**Comment by abarth** [comment]
I'm not sure we actually compile enough of the engine on Travis to run the dart tests.

**Comment by jason-simmons** [comment]
The set -e will exit after a failed command

Added verification of a test that fails intentionally.

Moved the script out of Travis with the intent of running it on the buildbots.

**Comment by abarth** [comment]
LGTM

**Pull request merged**